### PR TITLE
fix(ps-select): re-applies stateChanges patch and fixes label

### LINKED ISSUE #139: ps-select/ ps-form-field: Mat-Label not visible until first click/focus or sits tight on top
Hi,

the ps-select component does not work properly together with the ps-form-field since [refactor(select): merged ps-select-data into ps-select ](https://github.com/prosoft-edv/components/commit/f368694d1a339e286777dae8a79986acdb1ac089).

Either the label is not visible or it sits tight on top.
demo

# Steps for reproduction
1. Add PsFormBaseModule.forRoot(DemoPsFormsService) and PsFormFieldModule to select-demo.module.ts
2. Replace mat-form-field's in select-with-custom-select-service.component.ts with ps-form-field

### PR DESCRIPTION
**Pull request by schoenj**
Mat-Label was not visible until first/click or sits tight on top since last ps-select update.

fix #139

### CODE REVIEW

**Review by saithis — APPROVED** [review]

File: projects/components/select/src/select.component.ts [hunk 1]
```diff
@@ -89,6 +89,13 @@ export class PsSelectComponent<T = unknown> extends _PsSelectMixinBase
 
   @ViewChild(MatSelect, { static: true }) public set setMatSelect(select: MatSelect) {
     this._matSelect = select;
+
+    // MatSelect doesn't trigger stateChanges on close which causes problems, so we patch it here.
+    const close = select.close;
```
Comment (saithis) [inline projects/components/select/src/select.component.ts:94]: please add a test for the bug this fixes

File: projects/components/select/src/select.component.ts [hunk 2]
```diff
@@ -343,7 +350,7 @@ export class PsSelectComponent<T = unknown> extends _PsSelectMixinBase
 
   private _propagateValueChange(value: any, source: ValueChangeSource) {
     this._value = value;
-    this.empty = this.multiple ? !value?.length : value == null;
+    this.empty = this.multiple ? !value?.length : !value;
```
Comment (saithis) [inline projects/components/select/src/select.component.ts:353]: please add a test for the bug this fixes

**Review by saithis — CHANGES_REQUESTED** [review]

File: projects/components/select/src/select.component.ts (hunk 1 above)
Comment (schoenj) [reply projects/components/select/src/select.component.ts:94]: I enabled the existing test for that.

File: projects/components/select/src/select.component.ts (hunk 2 above)
Comment (schoenj) [reply projects/components/select/src/select.component.ts:353]: done

**Review by saithis — APPROVED** [review]

**Pull request merged**